src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -106,6 +109,26 @@ public void removePerson(Person key) {
 
     //// util methods
 
+    /**
+     * Gives the index of the person with the given name if they exist.
+     * If person doesn't exist, final index value is saved as length of the list.
+     * @param personList A list of person objects stored in data.
+     * @param name The given name used as reference to find the person.
+     * @return An Index object holding the index of the person in the list.
+     */
+    public Index getPersonListIndex(List<Person> personList, Name name) {
+        Index result = Index.fromZeroBased(0);
+        String personName = name.fullName;
+        for (Person i : personList) {
+            String currName = i.getName().fullName;
+            if (StringUtil.containsWordIgnoreCase(currName, personName)) {
+                break;
+            }
+            result.increment(1);
+        }
+        return result;
+    }
```

Comment:
Just checking, since the method containsWordIgnoreCase is used, we can delete e.g "Alice Pearson" by just typing the command of delete "Alice"? 
I am cool with this cause it can be convenient too :D

Reply:
Yes, it does. I thought it was nice to have. In the event of similar names, the first instance will be deleted. This will be resolved in the next iteration.

Reply:
To be precise, it deletes that person as long as the words match any part of their name. E.g.
`delete pearson` also works.

Reply:
Alright nice, this would fit for the future feature of clashing names found 👍🏽